Changed LoginForm

buckoon opened the pull request:
-Added TextField Components and conditioned them with state so the can perform error handling. Also, styled the Textfield components to match the Ui/Ux design.-Added Visibility/invisibility Icon to the password textFields on the register TextFields so the password can be hidden and unhidden by clicking the visibility icon.
-Added Check boxes from MUI and styled them in a similar manner
-Changed Font types and font sizes to make it look more professional
-implemented other changes on how the loginForm is effected in dark mode vs light mode. This was done using the useContext and state variables.

buckoon committed:
Updated the Login From with MUI components and styles:
-Added TextField Components and conditioned them with state so the can perform error handling. Also, styled the Textfield components to match the Ui/Ux design.-Added Visibility/invisibility Icon to the password textFields on the register TextFields so the password can be hidden and unhidden by clicking the visibility icon.
-Added Check boxes from MUI and styled them in a similar manner
-Changed Font types and font sizes to make it look more professional
-implemented other changes on how the loginForm is effected in dark mode vs light mode. This was done using the useContext and state variables.

buckoon committed:
Merge branch 'Librum-Reader:main' into main

DavidLazarescu commented:
Thanks for the PR, I'll be happy to merge it once you fix the following things:
- Reduce the space between the toolbar and the Login/Register components
![image](https://github.com/Librum-Reader/Librum-Website/assets/69865187/532ebc92-3d46-4d6c-bda6-d5f4804cb201)
- Make the color of the "Your credentials are only used to authenticate yourself. Everything will be stored in a secure database" text lighter, so that it is more visible. Keep it slightly darker than the actual "Welcome" text, so that it does **not** take the focus, but is still better visible.
- Make the "Welcome" and "Welcome back" titles a bit bigger
- Make the checkbox background (when selected) purple (#946BDE)
- Link the "privacy policy" and "Terms of service" links to the actual (temporary) pages at /termsofservice and /privacypolicy
- Make the whole "Don't have an account? Register" and "Already have an account? Login"  text clickable, not just the "Register" and "Login" parts of it
- Fix the centering of the place holder texts in the input fields (E.g. "Email" is 2 pixels too high). Make sure that everything is centered correctly
- Remove the "on-hover" effects for the input fields. (Don't change the border colors of the input fields when someone hovers over them)
- The marker on the "I accept the Terms of Service and the privacy policy" is not centered as shown in the picture below:
![image](https://github.com/Librum-Reader/Librum-Website/assets/69865187/7552e5af-64ed-4a88-8fdc-6314bbb97a50)
- Capitalize the "privacy policy" in "I accept the Terms of Service and the privacy policy" so that it says: "I accept the Terms of Service and the Privacy Policy"
- Decrease the spacing between the "Firstname" and "Lastname" input
![image](https://github.com/Librum-Reader/Librum-Website/assets/69865187/4d41daf0-07d9-4667-9075-0da92201fad4)

buckoon committed:
updated styling for Login UI

buckoon committed:
-Changed font size of headers over inputs from 16px to 14px
-Changed text color #aaa(darkmode font) to #CBCBCB
-Set a height for all input fields of 41.77px
-Setup Routes for Tos and Privacy Policy

buckoon committed:
-Added more spacing  in the vertical direction between the input fields
-Changed the color of the fonts and border of the input fields
